enhancements/microshift/feature-csi-snapshots.md

```diff
@@ -0,0 +1,566 @@
+9. The workload attaches the new volume and starts
+
+
+#### Volume Cloning
+
+Volume cloning uses CSI volume snapshotting interfaces behind the scenes.  It enables the pre-populating of a PVC's 
+storage volume with data from an existing PVC.
+
+_Prerequisites_
+
+- A PVC has been created following one of the provisioning workflows above.
+
+_Workflow_
+
+1. The user creates a PVC with the following fields:
+   - `spec.storageClassName`: must be the same storage class that provisioned the source volume
+   - `spec.dataSource.name`: name of the source `persistentVolumeClaim`
+   - `spec.dataSource.kind`: PersistentVolumeClaim
+   - `spec.dataSource.apiGroup`: core
+2. The user creates a Pod to consume the PVC. Required because LVMS only supports `WaitForFirstConsumer` provisioning.
+3. The _CSI external provisioner_ sidecar, already integrated into the _topolvm-controller_, detects the PVC request,
+   and makes a `CreateVolume` gRPC call to LVMS. 
+4. _topolvm-controller_ creates a `logicalVolume.topolvm.io` CRD with the same values of the source `logicalVolume.
+   topolvm.io` instance. This ensures the new volume is created on the same node as the source.
+5. _topolvm-node_ creates a new thin-volume from the volume snapshot.
+6. _topolvm-controller_ returns the success status to the `CreateVolume` gRPC caller.
+7. The _CSI external provisioner_ creates a `persistentVolume` to track the new thin-volume
+8. The _CSI external provisioner_ binds the PVC and PV together
+9. The workload attaches the new volume and starts
+
+
+#### Deploying
+
+The CSI Snapshot Controller and LVMS components are deployed by default on MicroShift. The manifests for these 
+components are baked into the MicroShift binary and are deployed upon startup. This follows the existing pattern 
+for MicroShift’s control-plane elements.
+
+The CSI Snapshotter configuration is managed via the VolumeSnapshotClass API. This API serves a similar purpose as 
+StorageClasses and allows admins to specify dynamic snapshotting parameters at runtime.  MicroShift will deploy a 
+default VolumeSnapshotClass on startup. This instance will reference the default StorageClass that is already 
+deployed by MicroShift to enable snapshotting out of the box.
+
+The MicroShift deployment model utilizes rpm-ostree layers. Following the existing deployment pattern, 
+VolumeSnapshotClass manifests can be packaged and deployed onto target devices.
+(See [kubernetes-for-device-edge. md#workflow-description.md](./kubernetes-for-device-edge.md#workflow-description)).
+Application deployers can predefine VolumeSnapshotClass manifests and use image-builder to package them into a 
+rpm-ostree layer.  This layer can be installed to the device, which writes the manifests to 
+/etc/microshift/manifests. This pattern is congruent with how a device owner would install custom StorageClasses. It 
+is recommended that new custom StorageClasses be packaged with VolumeSnapshotClasses that reference them.
```

Comment:
Here again, I wonder if we need to go into the details of how the manifests are deployed? The bottom line is that we suggest the users to put their manifests in `/etc/microshift/manifests` directory to be consumed by MicroShift on restart.